package-res/resources/web/pentaho/visual/base/model.js

```diff
@@ -45,37 +73,287 @@ define([
+         *
+         * @param {!pentaho.data.filter.AbstractFilter} proposedDataFilter -
+         * @param {!function} proposedSelectionMode -
+         * @return {pentaho.lang.ActionResult}
+         * If unsuccessful, the `error` field contains the exception that originated the error.
+         * If successful,  the `error` field is `null` and the `value` field contains the updated current selection.
+         *
+         * @fires "will:change:select"
+         * @fires "did:change:select"
+         * @fires "rejected:change:select"
+         *
+         * @see #selectAction
+         * @see #_runPipeline
+         * @see #_broadcastResult
+         * @protected
+         */
+        _changeSelection: function(proposedDataFilter, proposedSelectionMode) {
+          var selectionMode = proposedSelectionMode || this.getv("selectionMode");
+          var dataFilter = proposedDataFilter;
+
+          var result = this._runPipeline(WillChangeSelection, RejectedChangeSelection, dataFilter, selectionMode);
+          if(result) {
+            if(result.error) return result;
+            selectionMode = result.value.selectionMode;
+            dataFilter = result.value.dataFilter;
           }
-        })
-        .implement({meta: bundle.structured});
+
+          try {
+            var currentFilter = this.get("selectionFilter");
+            var combinedSelection = selectionMode(currentFilter, dataFilter);
+            if(combinedSelection) {
+              this.set("selectionFilter", combinedSelection);
+              result = new ActionResult(combinedSelection, null);
+            } else {
+              result = new ActionResult(undefined, new UserError(bundle.structured.error.selection.invalid));
+            }
+          } catch(e) {
+            result = new ActionResult(undefined, e);
+          }
+
+          return this._broadcastResult(DidChangeSelection, RejectedChangeSelection, result, dataFilter);
+        },
+
+        executeAction: function(proposedDataFilter) {
+          var doExecute = this.getv("doExecute");
+          var dataFilter = proposedDataFilter;
+
+          var result = this._runPipeline(WillExecute, RejectedExecute, dataFilter, doExecute);
+          if(result) {
+            if(result.error) return result;
+            doExecute = result.value.doExecute;
+            dataFilter = result.value.dataFilter;
+          }
+
+          if(doExecute) {
+            try {
+              doExecute(dataFilter);
+              result = new ActionResult(undefined, null);
+            } catch(e) {
+              result = new ActionResult(undefined, e);
+            }
+          } else {
+            result = new ActionResult(undefined, new UserError(bundle.structured.error.action.notDefined));
+          }
+          return this._broadcastResult(DidExecute, RejectedExecute, result, dataFilter);
+
+        },
+
+        /**
+         * Emits an event. All exceptions are caught (and swallowed).
```

Comment:
Hum... EventSource is more granular than the Action concept, so I don't think the will/did/reject things should go there.
Unfortunately, it looks like the comments are out of the original place, in the thread, so I'm not getting what exaclty you are referring to in the remainder of the text.